### ISSUE TITLE
Update `rimraf` to version 5

### ISSUE THREAD

**Issue by timvandermeij** [mozilla/pdf.js#16337]
In PR #15925 `rimraf` was updated from version 3 to version 4, but unfortunately this caused some breakage due to the removed glob functionaly (see https://github.com/mozilla/pdf.js/pull/15925#issuecomment-1455106763) and was consequently reverted in #16118.

However, in `rimraf` 4.2 the glob functionality was restored (see https://github.com/isaacs/rimraf/blob/main/CHANGELOG.md and https://github.com/isaacs/rimraf/issues/249#issuecomment-1396328533), so this update can be reattempted. Note that version 5 only removed the default export, so we might as well go to that newest version directory from version 3.

/cc @Snuffleupagus in case you're interested since you provided the previous PR which can hopefully be made to work now by providing the `glob` option to the places where we use globs.

**Comment by Snuffleupagus** [comment]
Given that the previous update attempt broke the release-process, i.e. the `gulp dist` part, I'd suggest that we don't try to update `rimraf` again until the releases are sorted out first.

@timvandermeij To improve the release-process, how feasible would https://github.com/mozilla/pdf.js/issues/11851#issuecomment-1493341576 be and would you perhaps be able to look into that?

**Comment by timvandermeij** [comment]
Yes, that one is next on my list since it's indeed becoming more important to get at least the first steps done there given the flaky release process lately. I hope I'll have time for this in the coming week.

**Comment by ojuschugh1** [comment]
Hi,
Is this issue still open?

**Comment by Snuffleupagus** [comment]
> Is this issue still open?

As outlined in https://github.com/mozilla/pdf.js/issues/16337#issuecomment-1519056246 this is currently blocked on other work being completed first (and given the trouble this gave us previously it's very unlikely to be a good beginner bug unfortunately).

**Comment by ojuschugh1** [comment]
ohh thanks for letting me @Snuffleupagus .

**Comment by timvandermeij** [comment]
Storing a reference to #17696 here as it might be an alternative to `rimraf`, and the new versions of `rimraf` also seem to simply re-export the Node.js ones already looking at https://github.com/isaacs/rimraf/blob/d2fa213e3ea1cc056698f9cac648b6e68796ea36/src/fs.ts#L5-L15 and https://github.com/isaacs/rimraf/issues/224#issuecomment-1381159636.

**Comment by Snuffleupagus** [comment]
> Storing a reference to #17696 here as it might be an alternative to get rid of `rimraf` altogether,

Slightly unrelated, but getting rid of `mkdirp` is probably almost trivial by comparison. (Although it will obviously require very *careful* manual testing during development.)

**Comment by timvandermeij** [comment]
Yes, I have a patch ready for that one already based on the PR above (obviously with the original author credited in the commit). I'm currently testing all Gulp targets affected by the change to make sure there are no differences before/after.

**Comment by wojtekmaj** [comment]
I guess this could be closed now :)

**Comment by timvandermeij** [comment]
Almost, but we still need to remove `rimraf` from the Gulpfile. I deliberately removed it from the test code first to keep the steps small and easier to test, given the complications we've had with `rimraf` before as mentioned in the original description above. But indeed, removing it is preferred over upgrading it now, and we're almost there :-)
